server/src/main/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapper.java

```diff
@@ -727,4 +730,40 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
     public FieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName()).init(this);
     }
+
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        return syntheticFieldLoader(simpleName());
+    }
+
+    protected SourceLoader.SyntheticFieldLoader syntheticFieldLoader(String simpleName) {
+        if (hasScript()) {
+            return SourceLoader.SyntheticFieldLoader.NOTHING;
+        }
+        if (fieldType().hasDocValues()) {
+            return new FlattenedSortedSetDocValuesSyntheticFieldLoader(name() + "._keyed", simpleName) {
+
+                @Override
+                protected BytesRef convert(BytesRef value) {
+                    byte[] bytes = Arrays.copyOfRange(value.bytes, value.offset, value.offset + value.length);
+                    String v = new String(bytes, StandardCharsets.UTF_8);
+                    return value;
```

Comment:
`value.utf8ToString()` should do this without the extra copy.

Reply:
I'd probably move these methods into `FlattenedSortedSetDocValuesSyntheticFieldLoader` unless we have a good reason not to.

Reply:
Sorry I did this for debugging and printing the string and forgot to remove it.

Reply:
About moving the methods...after removing the BytesRef to String stuff I realised we don't need to override.